src/main/java/frc/robot/subsystems/hood/HoodIOReal.java

```diff
@@ -0,0 +1,49 @@
+package frc.robot.subsystems.hood;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicExpoTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+
+public class HoodIOReal implements HoodIO {
+    private final TalonFX motor;
+    private final DutyCycleEncoder absoluteEncoder = new DutyCycleEncoder(9); // TODO: to be changed
+    private final MotionMagicExpoTorqueCurrentFOC positionControl =
+            new MotionMagicExpoTorqueCurrentFOC(0);
+    private final DutyCycleOut dutyCycleOut = new DutyCycleOut(0);
```

Comment:
Voltage or TorqueCurrentFOC are more accurate. The differences are listed here: https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/device-specific/talonfx/talonfx-control-intro.html

Reply:
45946ec

Reply:
a751780
@vichik123 said to use DutyCycle
6ab0d00

Reply:
@katzuv the link you sent is for closed loop control. Setting a certain output is open loop.

Reply:
OK, right.